src/main/java/org/webshar/hrms/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,27 @@
+package org.webshar.hrms.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import org.springframework.stereotype.Repository;
+import org.webshar.hrms.dao.RoleRepository;
+import org.webshar.hrms.dao.RoleRepositoryCustom;
+import org.webshar.hrms.models.db.Role;
+
+@Repository
+public class RoleDaoImpl implements RoleRepositoryCustom
```

Comment:
RoleRepositoryCustomImpl as the name will be more suitable here that RoleDaoImpl